### PR TITLE
Heterogenous comparisons

### PR DESCRIPTION
**Pull request by codeinred**
This merger is to implement comparisons of heterogenous types. For example:

```cpp
#include <string>

using tuplet::tuple;

bool do_stuff() {
    tuple<int, const char*> t1 { 10, "Hello" };
    tuple<int, std::string> t2 { 10, std::string("World") };
    // Comparing t1 and t2 is possible even though they're different types
    return t1 < t2; // Returns true because "Hello" comes before "World"
}
```

### CODE REVIEW

**Comment by codeinred** [comment]
@mechacrash Could I get your feedback on this code? It adds heterogenous comparisons to tuple so that you can compare tuples of different types. Also, would you know any way to get it to work in clang 10 or 11? I had to disable the tests for heterogenous comparisons in those versions of clang :(

**Comment by mechacrash** [comment]
> I threw together a quick alternative method (converting the input tuple into a new instance that matches types exactly, 

yeah, I clearly wasn’t thinking very hard when I did this - of course this won’t work for types that don’t support construction from one another 🙃

**Comment by mechacrash** [comment]
Yeah, I’m gonna continue tinkering too! This project inspired me to revisiting my old triple with a fresh perspective and some new ideas, and I’ve managed some big improvements! I think there’ll be stuff I can bring back here too 😁

being added as a contributor would be great, sure! I can’t promise I’ll be able to do all that much, but I’ll do my best!

Also… another idea I had right before going to bed.
We don’t _have_ to use <=> right? We could just write code for its constituent parts… then it wouldn’t need the overloaded ^ operator to handle compound <=> results, which is what clang is fumbling here.
Would be more code, and worse debug build codegen… but it would probably compile! 😅

File: include/tuplet/tuple.hpp [hunk 1]
```diff
@@ -78,6 +91,11 @@ template <class... Ls, class... Rs>
 constexpr auto operator+(type_list<Ls...>, type_list<Rs...>) {
     return type_list<Ls..., Rs...> {};
 }
+template <class T, class U>
+constexpr auto operator^(T a, U b) noexcept
```
Comment (NN---) [inline include/tuplet/tuple.hpp:95]: Is it really required to add such operator ?
Why not a function ?

**Comment by NN---** [comment]
Tried to compile using MSVC and got an error. Please vote for the fix.

https://developercommunity.visualstudio.com/t/Wrong-error:-C2120:-void-illegal-with/1678810

**Comment by codeinred** [comment]
Resolved as of v2.0.0 - tuplet now supports heterogenous comparisons on all compilers, including MSVC!

**Pull request closed without merging**